packages/helm-charts/testnet/templates/validators.statefulset.yaml

```diff
@@ -61,20 +61,56 @@ spec:
             celotooljs.sh generate bip32 --mnemonic "$MNEMONIC" --accountType validator --index $RID > /root/.celo/pkey
             echo 'Generating address'
             celotooljs.sh generate account-address --private-key `cat /root/.celo/pkey` > /root/.celo/address
-            if [ -z $IP_ADDRESSES ]; then
+            echo -n $IP_ADDRESSES | cut -d '/' -f $((RID + 1)) > /root/.celo/externalIpAddress
+
+            if [[ -z $IP_ADDRESSES || "$RID" -lt "{{ .Values.geth.proxiedValidators }}" ]]; then
               echo $POD_IP > /root/.celo/ipAddress
             else
-              echo $IP_ADDRESSES | cut -d '/' -f $((RID + 1)) > /root/.celo/ipAddress
+              cat /root/.celo/externalIpAddress > /root/.celo/ipAddress
             fi
             echo -n "Generating IP address for validator: "
             cat /root/.celo/ipAddress
+
             celotooljs.sh generate public-key --mnemonic "$MNEMONIC" --accountType load_testing --index 0 > /root/.celo/bootnodeEnodeAddress
             echo -n "Generating Bootnode enode address for the validator: "
             cat /root/.celo/bootnodeEnodeAddress
+
             [[ "$BOOTNODE_IP_ADDRESS" == 'none' ]] && BOOTNODE_IP_ADDRESS=${{ .Release.Namespace | upper }}_BOOTNODE_SERVICE_HOST
             echo `cat /root/.celo/bootnodeEnodeAddress`@$BOOTNODE_IP_ADDRESS:30301 > /root/.celo/bootnodeEnode
             echo -n "Generating Bootnode enode for the validator: "
             cat /root/.celo/bootnodeEnode
+
```

Comment:
This is becoming a pretty sizable script, consider extracting it into a proper files and then using helm to inject it